### PR TITLE
add barbican quota implementation

### PR DESCRIPTION
**Pull request by snigle**
Fixes gophercloud#2301

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://docs.openstack.org/barbican/4.0.0/api/reference/quotas.html
https://github.com/openstack/barbican/blob/master/barbican/api/controllers/quotas.py

### CODE REVIEW

**Comment by snigle** [comment]
hello, I don't see any logs about the failed acceptance test, can you help ?

**Comment by pierreprinetti** [comment]
recheck

**Comment by EmilienM** [comment]
Hey folks, I'm going to add CI coverage for Barbican. Hold on and we'll be able to test this PR and merge it if it's all good.

Thanks

**Comment by EmilienM** [comment]
Here we go: https://github.com/gophercloud/gophercloud/pull/2327
once we have this, let's rebase this PR on master then we'll have CI covering this PR.

Thanks for your patience and your contribution :)

**Comment by EmilienM** [comment]
@snigle please rebase your PR on master, thanks

**Comment by EmilienM** [comment]
I don't understand why the new CI job hasn't run. Let me check.

**Comment by EmilienM** [comment]
@snigle sorry, we did something wrong and will need https://github.com/gophercloud/gophercloud/pull/2334 to land. Once this lands, we should be able to retrigger the keymanager CI job. Thanks again for your patience :)

File: acceptance/openstack/keymanager/v1/quotas_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/acceptance/clients"
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/projects"
+	"github.com/gophercloud/gophercloud/openstack/keymanager/v1/quotas"
+	th "github.com/gophercloud/gophercloud/testhelper"
+)
+
+func TestQuotasCRUD(t *testing.T) {
+	clients.RequireAdmin(t)
```
Comment (EmilienM) [inline acceptance/openstack/keymanager/v1/quotas_test.go:17]: weird, this test is failing despite you're an admin:

=== RUN   TestQuotasCRUD
    convenience.go:35: Failure in quotas_test.go, line 46: unexpected error "Request forbidden: [PUT http://10.1.0.42/key-manager/v1/project-quotas/052045906ba04ff596538b79f9bc8288], error message: {\"code\": 403, \"title\": \"Forbidden\", \"description\": \"Project Quotas attempt not allowed - please review your user/project privileges\"}"

File: acceptance/openstack/keymanager/v1/quotas_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+	clients.SkipRelease(t, "stable/mitaka")
+	clients.SkipRelease(t, "stable/newton")
+	clients.SkipRelease(t, "stable/queens")
```
Comment (EmilienM) [inline acceptance/openstack/keymanager/v1/quotas_test.go:21]: Please remove the skips, we don't need them. Our CI covers down to Train only.

**Comment by snigle** [comment]
Hello @EmilienM , I still have issue with tests :'(
Can you take back the code to check in development environment ?

**Comment by snigle** [comment]
looks like your administrator user doesn't have the right to update quotas ?

**Comment by EmilienM** [comment]
> looks like your administrator user doesn't have the right to update quotas ?

Yeah I saw. I'll take a look when time permits. If you want to debug, you can download the job artifacts here: https://github.com/gophercloud/gophercloud/actions/runs/5974521913?pr=2302 and in journal file you should see all OpenStack logs.

**Comment by EmilienM** [comment]
This PR has been staled for a very long time, I'll close it. Feel free to re-open!

**Comment by snigle** [comment]
Hello, we still need it, we use it in production on our fork.

**Pull request closed without merging**